pennylane_lightning/lightning_base.py

```diff
@@ -16,190 +16,398 @@
+    _CPP_BINARY_AVAILABLE = True
+
+    def __init__(
+        self,
+        wires,
+        *,
+        c_dtype=np.complex128,
+        shots=None,
+        batch_obs=False,
+    ):
+        if c_dtype is np.complex64:
+            r_dtype = np.float32
+            self.use_csingle = True
+        elif c_dtype is np.complex128:
+            r_dtype = np.float64
+            self.use_csingle = False
+        else:
+            raise TypeError(f"Unsupported complex Type: {c_dtype}")
+        super().__init__(wires, shots=shots, r_dtype=r_dtype, c_dtype=c_dtype)
+        self._batch_obs = batch_obs
+
+    @property
+    def stopping_condition(self):
+        """.BooleanFn: Returns the stopping condition for the device. The returned
+        function accepts a queueable object (including a PennyLane operation
+        and observable) and returns ``True`` if supported by the device."""
+
+        def accepts_obj(obj):
+            if obj.name == "QFT" and len(obj.wires) < 10:
+                return True
+            if obj.name == "GroverOperator" and len(obj.wires) < 13:
+                return True
+            return (not isinstance(obj, qml.tape.QuantumTape)) and getattr(
+                self, "supports_operation", lambda name: False
+            )(obj.name)
+
+        return qml.BooleanFn(accepts_obj)
+
+    @classmethod
+    def capabilities(cls):
+        capabilities = super().capabilities().copy()
+        capabilities.update(
+            model="qubit",
+            supports_analytic_computation=True,
+            supports_broadcasting=False,
+            returns_state=True,
+        )
+        return capabilities
+
+    # To be able to validate the adjoint method [_validate_adjoint_method(device)],
+    #  the qnode requires the definition of:
+    # ["_apply_operation", "_apply_unitary", "adjoint_jacobian"]
+    # pylint: disable=missing-function-docstring
+    def _apply_operation(self):
+        pass
+
+    # pylint: disable=missing-function-docstring
+    def _apply_unitary(self):
+        pass
+
+    def _init_process_jacobian_tape(self, tape, starting_state, use_device_state):
+        """Generate an initial state vector for ``_process_jacobian_tape``."""
+
+    @property
+    def create_ops_list(self):
+        """Returns create_ops_list function of the matching precision."""
+
+    def probability_lightning(self, wires):
+        """Return the probability of each computational basis state."""
+
+    def vjp(self, measurements, grad_vec, starting_state=None, use_device_state=False):
+        """Generate the processing function required to compute the vector-Jacobian
+        products of a tape.
+        """
 
-try:
-    from .pennylane_lightning_ops import (
-        MeasurementsC64,
-        StateVectorC64,
-        MeasurementsC128,
-        StateVectorC128,
-        backend_info,
-    )
+    def probability(self, wires=None, shot_range=None, bin_size=None):
+        """Return the probability of each computational basis state.
```

Comment:
Can we add `pass` to all of these? Or return `None` if `pass` isn't acceptable for `property`

Reply:
This was from [pylint](https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/unnecessary-pass.html). I don't mind either way.

Reply:
Hmm, I'm not sure I agree with pylint on this one, but in the interests of moving along I'll retract my request